server/models/base/Model.ts

```diff
@@ -63,6 +65,13 @@ class Model<
     return this.destroy(ctx.context as InstanceDestroyOptions);
   }
 
+  /**
+   * Restore the row corresponding to this instance. Only available for paranoid models.
+   */
+  public restoreWithCtx(ctx: APIContext) {
```

Comment:
`changeset` is empty for restore flow ([sequelize open issue](https://github.com/sequelize/sequelize/issues/12912)). This means the `changes` will not be populated in the published event. I think this should be fine in this case - we can always manually build it if the need arises.